src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -190,7 +190,8 @@ protected async Task<AuthenticationResult> CacheTokenResponseAndCreateAuthentica
 
             if (!AuthenticationRequestParameters.IsClientCredentialRequest &&
                 AuthenticationRequestParameters.ApiId != ApiEvent.ApiIds.AcquireTokenByRefreshToken &&
-                AuthenticationRequestParameters.AuthorityInfo.AuthorityType != AuthorityType.Adfs)
+                AuthenticationRequestParameters.AuthorityInfo.AuthorityType != AuthorityType.Adfs &&
+                !(msalTokenResponse.ClientInfo is null))
```

Comment:
We need to add some logging statements to make sure we understand we're dealing with SP OBO and not normal OBO, if we read the MSAL's logs. Perhaps in the OBO request class ...